docs: add model validation step (#18)
GoogleCloudPlatform/terraform-google-enterprise-genai: 6-ml-pipeline/dev/Readme.md

### 4. Merge and deploy
- Once everything is configured, you can commit your changes and push to the dev branch. Then, create a PR to from dev to staging(non-prod) which will result in triggering the pipeline if approved. The vertex pipeline takes about 30 minutes to finish and if there are no errors, a trained model will be deployed to and endpoint in the prod project which you can use to make prediction requests.

### 5. Model Validation

> - Once you have the model running at an endpoint in the production project, you will be able to test it. Here is a step-by-step to make a request to your model using `gcloud` and `curl`:

Reviewer: ```suggestion
Once you have the model running at an endpoint in the production project, you will be able to test it.
Here are step-by-step instructions to make a request to your model using `gcloud` and `curl`:
```

1. Initialize variables on your terminal session

>     ```bash
>     ENDPOINT_ID=<REPLACE_WITH_ENDPOINT_ID>
>     PROJECT_ID=<REPLACE_WITH_PROJECT_ID>
>     INPUT_DATA_FILE="body.json"
>     ```
>     > You can retrieve your ENDPOINT_ID by running `gcloud ai endpoints list --region=us-central1 --project=<PROD_ML_PROJECT>` or by navigating to it on the google cloud console

Reviewer: ```suggestion
    ```bash
    ENDPOINT_ID=<REPLACE_WITH_ENDPOINT_ID>
    PROJECT_ID=<REPLACE_WITH_PROJECT_ID>
    INPUT_DATA_FILE="body.json"
    ```

    > You can retrieve your ENDPOINT_ID by running `gcloud ai endpoints list --region=us-central1 --project=<PROD_ML_PROJECT>` or by navigating to it on the google cloud console
```

add one empty line around fences

>     ```bash
>     ENDPOINT_ID=<REPLACE_WITH_ENDPOINT_ID>
>     PROJECT_ID=<REPLACE_WITH_PROJECT_ID>
>     INPUT_DATA_FILE="body.json"
>     ```
>     > You can retrieve your ENDPOINT_ID by running `gcloud ai endpoints list --region=us-central1 --project=<PROD_ML_PROJECT>` or by navigating to it on the google cloud console

Reviewer: ```suggestion
    ```bash
    ENDPOINT_ID=<REPLACE_WITH_ENDPOINT_ID>
    PROJECT_ID=<REPLACE_WITH_PROJECT_ID>
    INPUT_DATA_FILE="body.json"
    ```
    > You can retrieve your ENDPOINT_ID by running `gcloud ai endpoints list --region=us-central1 --project=<PROD_ML_PROJECT>` or by navigating to it on the Google Cloud Console (https://console.cloud.google.com/vertex-ai/online-prediction/endpointshttps://console.cloud.google.com/vertex-ai/online-prediction/endpoints?project=<PROD_ML_PROJECT>`)
```

2. Create a file named `body.json` and put some sample data into it:

    ```json
    {
        "instances": [
            {
                "features/gender": "Female",
                "features/workclass": "Private",
                "features/occupation": "Tech-support",
                "features/marital_status": "Married-civ-spouse",
                "features/race": "White",
                "features/capital_gain": 0,
                "features/education": "9th",
                "features/age": 33,
                "features/hours_per_week": 40,
                "features/relationship": "Wife",
                "features/native_country": "Canada",
                "features/capital_loss": 0
            }
        ]
    }
    ```

3. Run a curl request using `body.json` file as the JSON Body.

>     ```bash
>     curl -X POST -H "Authorization: Bearer $(gcloud auth print-access-token)" -H "Content-Type: application/json" https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/endpoints/${ENDPOINT_ID}:predict -d "@${INPUT_DATA_FILE}"
>     ```

Reviewer: ```suggestion
    ```bash
    curl -X POST -H "Authorization: Bearer $(gcloud auth print-access-token)" \
    -H "Content-Type: application/json" \
    https://us-central1-aiplatform.googleapis.com/v1/projects/${PROJECT_ID}/locations/us-central1/endpoints/${ENDPOINT_ID}:predict -d "@${INPUT_DATA_FILE}"
    ```
```

>     - You should get an output from 0 to 1, indicating the level of confidence of the binary classification based on the parameters above. Values closer to 1 means the individual is more likely to be included in the income_bracket greater than 50K.

Reviewer: ```suggestion
    - You should get an output from 0 to 1, indicating the level of confidence of the binary classification based on the parameters above.
    Values closer to 1 means the individual is more likely to be included in the income_bracket greater than 50K.
```
